src/main/java/plugily/projects/villagedefense/arena/managers/ScoreboardManager.java

```diff
@@ -43,7 +47,9 @@
  */
 public class ScoreboardManager {
 
-  private final List<Scoreboard> scoreboards = new ArrayList<>();
+  private final Map<UUID, Scoreboard> boardMap = new ConcurrentHashMap<>();
+  private final Map<UUID, org.bukkit.scoreboard.Scoreboard> lastBoardMap = new ConcurrentHashMap<>();
```

Comment:
You literally created two `HashMap` instance instead of storing in `User` object. This is not optimization.

Reply:
For loop in remove scoreboard is also not a optimized way

Reply:
> You literally created two `HashMap` instance instead of storing in `User` object. This is not optimization.

And it stores the unique id, not the whole User object

Reply:
It's better because you guys can add an option to enable/disable the scoreboard manager.

Reply:
Neither is better, but it doesn't matter at all.

Reply:
Well... I just don't care. If it's not an optimization, it will be a small refactor.